streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredTimestampedKeyValueStore.java

```diff
@@ -53,4 +56,28 @@ void initStoreSerde(final ProcessorContext context) {
             keySerde == null ? (Serde<K>) context.keySerde() : keySerde,
             valueSerde == null ? new ValueAndTimestampSerde<>((Serde<V>) context.valueSerde()) : valueSerde);
     }
+
+    public RawAndDeserializedValue<V> getWithBinary(final K key) {
+        try {
+            final byte[] serializedValue = wrapped().get(keyBytes(key));
+            return new RawAndDeserializedValue<V>(serializedValue,
+                maybeMeasureLatency(() -> outerValue(serializedValue), time, getSensor));
+        } catch (final ProcessorStateException e) {
+            final String message = String.format(e.getMessage(), key);
+            throw new ProcessorStateException(message, e);
+        }
+    }
+
+    public boolean putIfDifferent(final K key,
+                                  final ValueAndTimestamp<V> newValue,
+                                  final byte[] oldSerializedValue) {
+        final Bytes serializedNewValueBytes = Bytes.wrap(serdes.rawValue(newValue));
+        final Bytes serializedOldValueBytes = Bytes.wrap(oldSerializedValue);
+        if (serializedNewValueBytes == null ||
+            !serializedNewValueBytes.equals(serializedOldValueBytes)) {
+            super.put(key, newValue);
+            return true;
+        }
+        return false;
+    }
```

Comment:
```suggestion
    public boolean putIfDifferentValues(final K key,
                                        final ValueAndTimestamp<V> newValue,
                                        final byte[] oldSerializedValue) {
        try {
            return maybeMeasureLatency(
                () -> {
                    final byte[] newSerializedValue = serdes.rawValue(newValue);
                    if (ValueAndTimestampSerializer.maskTimestampAndCompareValues(oldSerializedValue, newSerializedValue)) {
                        return false;
                    } else {
                        wrapped().put(keyBytes(key), newSerializedValue);
                        return true;
                    }
                },
                time,
                putSensor
            );
        } catch (final ProcessorStateException e) {
            final String message = String.format(e.getMessage(), key, newValue);
            throw new ProcessorStateException(message, e);
        }
    }
```

Recommending a few tweaks here as well:
1. We need to measure the put latency including serialization
2. We should avoid serializing twice (which happens when you delegate to `super.put`)
3. For KTables at least, we should ignore the timestamp when deciding whether to really put it or not, so I renamed the method to clarify it's only comparing the value, and I also created an extra comparison method in the `ValueAndTimestampSerializer` (since it manages the schema, it can safely skip over the timestamp in the serialized data).

Here's what I sketched for the ValueAndTimestampSerializer method:
```java
    public static boolean maskTimestampAndCompareValues(final byte[] left, final byte[] right) {
        // adapted from Arrays.equals
        if (left == right)
            return true;
        if (left ==null || right ==null)
            return false;

        final int length = left.length;
        if (right.length != length)
            return false;

        // skip the timestamp when comparing just the values
        for (int i=Long.BYTES; i<length; i++)
            if (left[i] != right[i])
                return false;

        return true;
    }
```